### PR TITLE
fix: simplify fully-qualified image names

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
This PR removes all the prefixes for `docker.io` from the images used in 1) tests for the library, 2) modules

## Why is it important?
It's better to ask for "nginx:latest" and let the engine deal with the image references than to enforce an invalid docker registry.

If we wanted to be explicit about official images, we should have used `https://index.docker.io/v1/library/nginx:latest` instead.

It also removes a bug when using Docker Desktop, as it considered those images were not accessible for the already used docker credentials.

### CODE REVIEW

**Review by kiview — DISMISSED** [review]

File: container_test.go [hunk 1]
```diff
@@ -392,11 +392,11 @@ func Test_GetLogsFromFailedContainer(t *testing.T) {
 type dockerImageSubstitutor struct{}
 
 func (s dockerImageSubstitutor) Description() string {
-	return "DockerImageSubstitutor (prepends docker.io)"
+	return "DockerImageSubstitutor (prepends registry.hub.docker.com)"
 }
 
 func (s dockerImageSubstitutor) Substitute(image string) (string, error) {
-	return "docker.io/" + image, nil
+	return "registry.hub.docker.com/library/" + image, nil
```
Comment (stevenh) [inline container_test.go:399]: question: should we have an private constant for this?

File: docker_files_test.go [hunk 2]
```diff
@@ -66,7 +66,7 @@ func TestCopyFileToRunningContainer(t *testing.T) {
 
 	ctr, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
 		ContainerRequest: testcontainers.ContainerRequest{
-			Image: "docker.io/bash:5.2.26",
+			Image: "bash:5.2.26",
```
Comment (stevenh) [inline docker_files_test.go:69]: question: others don't have a version, why is this one special?

File: modules/postgres/postgres_test.go [hunk 3]
```diff
@@ -303,7 +303,7 @@ func TestSnapshotWithOverrides(t *testing.T) {
 
 	ctr, err := postgres.Run(
 		ctx,
-		"docker.io/postgres:16-alpine",
+		"6-alpine",
```
Comment (stevenh) [inline modules/postgres/postgres_test.go:306]: bug: typo
```suggestion
		"postgres:16-alpine",
```

**Comment by stevenh** [comment]
Oh if the concern is about *.io going away what about https://index.docker.io/v1/?

**Comment by mdelapenya** [comment]
> Oh if the concern is about *.io going away what about https://index.docker.io/v1/?

No, because that's the correct index for the registry (Docker Hub)

File: docker_files_test.go (hunk 2 above)
Comment (mdelapenya) [reply docker_files_test.go:69]: No special at all, I wonder why it was not included for all the occurrences. I think we can change that in this PR

File: docker_files_test.go [hunk 4]
```diff
@@ -30,7 +30,7 @@ func TestCopyFileToContainer(t *testing.T) {
 
 	ctr, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
 		ContainerRequest: testcontainers.ContainerRequest{
-			Image: "docker.io/bash",
+			Image: "bash:5.2.26",
```
Comment (stevenh) [inline docker_files_test.go:33]: suggestion: we should add constant so its easy to update. We might even consider just using :latest given what it is.

File: docker_test.go [hunk 5]
```diff
@@ -950,7 +950,7 @@ func TestWorkingDir(t *testing.T) {
 func ExampleDockerProvider_CreateContainer() {
 	ctx := context.Background()
 	req := ContainerRequest{
-		Image:        "docker.io/nginx:alpine",
+		Image:        "nginx:alpine",
```
Comment (stevenh) [inline docker_test.go:953]: suggestion: use the existing constant nginxAlpineImage, more below

**Review by stevenh — CHANGES_REQUESTED** [review]
Looking good, just a few suggestions while you're making changes.

File: docker_test.go (hunk 5 above)
Comment (mdelapenya) [reply docker_test.go:953]: I did not do it on purpose, as the testable examples could eventually land into a docs page, like pkg.go.dev, and the variable won't be interpolated.

**Review by stevenh — APPROVED** [review]

**Pull request merged**